visualization/app/codeCharta/ui/codeMap/codeMap.actions.service.ts

```diff
@@ -48,34 +48,45 @@ export class CodeMapActionsService {
 		const { edges } = state.fileSettings
 		const { edgeMetric } = state.dynamicSettings
 		const numberOfEdgesToDisplay = state.appSettings.amountOfEdgePreviews
-		const edgePreviewNodes = this.edgeMetricDataService.getNodesWithHighestValue(edgeMetric, numberOfEdgesToDisplay)
-
-		edges.forEach(edge => {
-			if (
-				(edgePreviewNodes.includes(edge.fromNodeName) || edgePreviewNodes.includes(edge.toNodeName)) &&
-				Object.keys(edge.attributes).includes(edgeMetric)
-			) {
-				edge.visible = EdgeVisibility.both
-				if (!edgePreviewNodes.includes(edge.fromNodeName)) {
-					edge.visible = EdgeVisibility.to
-				} else if (!edgePreviewNodes.includes(edge.toNodeName)) {
-					edge.visible = EdgeVisibility.from
+		const edgePreviewNodes = new Set(this.edgeMetricDataService.getNodesWithHighestValue(edgeMetric, numberOfEdgesToDisplay))
+
+		for (const edge of edges) {
+			if (Object.prototype.hasOwnProperty.call(edge.attributes, edgeMetric)) {
```

Comment:
```suggestion
			if (edge.attributes[edgeMetric] !== undefined)) {
```

Same suggestion as in another comment.

Reply:
The current one is a tad safer, but it's unlikely to be of any importance.